### PR TITLE
Add support for AWS_REGION environment variable to enhance multi-region functionality

### PR DESCRIPTION
**Pull request by bshutterkion**
Addresses Issue #3

### CODE REVIEW

File: lib/kion/stak.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ type STAK struct {
 	AccessKey       string `json:"access_key"`
 	SecretAccessKey string `json:"secret_access_key"`
 	SessionToken    string `json:"session_token"`
+	AWS_REGION      string `json:"aws_region"`
```
Comment (blt04) [inline lib/kion/stak.go:25]: This should be `AwsRegion` to follow our naming conventions.

File: main.go [hunk 2]
```diff
@@ -319,6 +325,19 @@ func genStaks(cCtx *cli.Context) error {
...
+
+	var defaultRegion string
+
+	// Now check the TypeID of the selected account
+	if selectedAccount.TypeID == 1 {
+		defaultRegion = userConfig.Data.AwsDefaultCommercialRegion
+	} else if selectedAccount.TypeID == 2 {
+		defaultRegion = userConfig.Data.AwsDefaultGovcloudRegion
+	} else {
+		return fmt.Errorf("unknown genStaks account type")
```
Comment (blt04) [inline main.go:338]: I don't think you need an else condition.  This will break when we support non-AWS account types.

File: main.go [hunk 3]
```diff
@@ -373,18 +392,35 @@ func genStaksFav(cCtx *cli.Context) error {
...
-	stak, err := kion.GetSTAK(cCtx.String("endpoint"), cCtx.String("token"), favorite.CAR, favorite.Account)
+
+	var defaultRegion string
+
+	if favorite.AccountType == 1 {
+		defaultRegion = userConfig.Data.AwsDefaultCommercialRegion
+	} else if favorite.AccountType == 2 {
+		defaultRegion = userConfig.Data.AwsDefaultGovcloudRegion
+	} else {
+		return fmt.Errorf("unknown genStaksFav account type: %d", favorite.AccountType)
```
Comment (blt04) [inline main.go:411]: Similar to the above, I think we can omit the else condition

File: main.go [hunk 4]
```diff
@@ -439,19 +475,41 @@ func runCommand(cCtx *cli.Context) error {
...
+		}
+
+		var defaultRegion string
+
+		if selectedAccount.AccountType == 1 {
+			defaultRegion = userConfig.Data.AwsDefaultCommercialRegion
+		} else if selectedAccount.AccountType == 2 {
+			defaultRegion = userConfig.Data.AwsDefaultGovcloudRegion
+		} else {
+			return fmt.Errorf("unknown fav account type")
```
Comment (blt04) [inline main.go:504]: One more, omit else

**Review by blt04 — CHANGES_REQUESTED** [review]

File: lib/kion/user_default_regions.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		return UserConfigResponse{}, err
+	}
+
+	// Check for null values and set defaults
+	if defaultRegionResp.Data.AwsDefaultCommercialRegion == "" {
+		defaultRegionResp.Data.AwsDefaultCommercialRegion = "us-east-1"
+	}
+	if defaultRegionResp.Data.AwsDefaultGovcloudRegion == "" {
+		defaultRegionResp.Data.AwsDefaultGovcloudRegion = "us-gov-east-1"
+	}
```
[the remark this answers is not in the document]
Comment (BrandonShutter) [reply lib/kion/user_default_regions.go:46]: If we don't set a default region, we're gonna run into the same situation that I ran into that prompted me to create this issue. If we don't set a default, we'll have to inform users in some other way, that they need to set their own region.

[the remark this answers is not in the document]
Comment (blt04) [reply lib/kion/user_default_regions.go:46]: So maybe we only do it for govcloud, but not commercial partition?

**Pull request closed without merging**